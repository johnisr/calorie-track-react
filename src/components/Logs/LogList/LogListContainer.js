import React from 'react';
import { connect } from 'react-redux';
import LogList from './LogList';
import { addCurrentEditLog } from '../../../actions/currentLog';
import { defaultLog, startAddLog } from '../../../actions/logs';
import selectLogsWithPages from '../../../selectors/logsWithPages';

export class LogListContainer extends React.Component {
  handleEdit = (date) => {
    const log = this.props.logs.filter(log => log.date === date)[0];
    this.props.addCurrentEditLog(log);
    this.props.history.push('/editlog');
  }
  handleCreate = (time) => {
    const date = time.startOf('day').valueOf();
    const log = this.props.logs.filter(log => log.date === date)[0];
    if (!log) {
      const newLog = {
        ...defaultLog,
        date,
      };
      this.props.startAddLog(newLog);
      this.props.addCurrentEditLog(newLog);
    } else {
      this.props.addCurrentEditLog(log);
    }
    this.props.history.push('/editlog');
  }
  render() {
    return (
      <LogList
        logs={this.props.logs}
        onEdit={(date) => this.handleEdit(date)}
        onCreate={(time) => this.handleCreate(time)}
      />
    );
  }
}  

const mapStateToProps = (state) => ({
  logs: selectLogsWithPages(state.logs, state.logsFilters),
});

const mapDispatchToProps = (dispatch) => ({
  startAddLog: (log) => dispatch(startAddLog(log)),
  addCurrentEditLog: (log) => dispatch(addCurrentEditLog(log)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LogListContainer);
