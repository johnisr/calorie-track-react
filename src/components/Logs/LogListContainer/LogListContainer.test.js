import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { LogListContainer } from './LogListContainer';
import logsData from '../../../tests/fixtures/logs';

let logs, startAddLog, addCurrentEditLog, history, wrapper;
beforeEach(() => {
  logs = logsData;
  startAddLog = jest.fn();
  addCurrentEditLog = jest.fn();
  history = { push: jest.fn() }
  wrapper = shallow(<LogListContainer
    logs={logs}
    startAddLog={startAddLog}
    addCurrentEditLog={addCurrentEditLog}
    history={history}
  />);
});

it('should render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handleEdit a log', () => {
  const date = logs[1].date;
  wrapper.find('LogList').prop('onEdit')(date);
  expect(addCurrentEditLog).toHaveBeenLastCalledWith(logs[1]);
  expect(history.push).toHaveBeenLastCalledWith('/editlog');
});

it('should edit instead of create a when given a date previously used', () => {
  wrapper.find('LogList').prop('onCreate')(moment(logs[0].date));
  expect(addCurrentEditLog).toHaveBeenLastCalledWith(logs[0]);
  expect(startAddLog).not.toHaveBeenCalled();
  expect(history.push).toHaveBeenCalledWith('/editlog');
});

it('should create a new time when given an new date', () => {
  const log = {
    date: moment('20000101', 'YYYYMMDD').valueOf(),
    weight: 0,
    unit: 'lb',
    foods: [],
    total: {
      protein: 0,
      carbohydrates: 0,
      fat: 0,
      calories: 0,
    },
  };
  wrapper.find('LogList').prop('onCreate')(moment('20000101', 'YYYYMMDD'));
  expect(startAddLog).toHaveBeenLastCalledWith(log);
  expect(addCurrentEditLog).toHaveBeenLastCalledWith(log);
  expect(history.push).toHaveBeenCalledWith('/editlog');
});