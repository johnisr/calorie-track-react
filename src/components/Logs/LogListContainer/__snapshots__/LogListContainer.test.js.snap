// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogListContainer
    addCurrentEditLog={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    logs={
        Array [
            Object {
              "date": -154800000,
              "foods": Array [
                Object {
                  "amount": 24,
                  "calories": 163,
                  "carbohydrates": 2.5,
                  "createdAt": -1000,
                  "fat": 14,
                  "id": "3",
                  "name": "Almonds",
                  "protein": 6,
                  "timesUsed": 2,
                  "unit": "pieces",
                },
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 100,
                  "base": Object {
                    "amount": 100,
                    "calories": 165,
                    "carbohydrates": 0,
                    "fat": 3.6,
                    "protein": 31,
                  },
                  "calories": 165,
                  "carbohydrates": 0,
                  "createdAt": 1000,
                  "fat": 3.6,
                  "id": "2",
                  "name": "Chicken Breast",
                  "protein": 31,
                  "timesUsed": 1,
                  "unit": "grams",
                },
              ],
              "total": Object {
                "calories": 533,
                "carbohydrates": 46.5,
                "fat": 18,
                "protein": 41.2,
              },
              "unit": "lb",
              "weight": 202,
            },
            Object {
              "date": 0,
              "foods": Array [
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
              ],
              "total": Object {
                "calories": 410,
                "carbohydrates": 88,
                "fat": 0.8,
                "protein": 8.4,
              },
              "unit": "lb",
              "weight": 200,
            },
            Object {
              "date": 86400000,
              "foods": Array [
                Object {
                  "amount": 200,
                  "calories": 205,
                  "carbohydrates": 44,
                  "createdAt": 0,
                  "fat": 0.4,
                  "id": "1",
                  "name": "Rice (Jasmine)",
                  "protein": 4.2,
                  "timesUsed": 0,
                  "unit": "grams",
                },
                Object {
                  "amount": 100,
                  "base": Object {
                    "amount": 100,
                    "calories": 165,
                    "carbohydrates": 0,
                    "fat": 3.6,
                    "protein": 31,
                  },
                  "calories": 165,
                  "carbohydrates": 0,
                  "createdAt": 1000,
                  "fat": 3.6,
                  "id": "2",
                  "name": "Chicken Breast",
                  "protein": 31,
                  "timesUsed": 1,
                  "unit": "grams",
                },
                Object {
                  "amount": 24,
                  "calories": 163,
                  "carbohydrates": 2.5,
                  "createdAt": -1000,
                  "fat": 14,
                  "id": "3",
                  "name": "Almonds",
                  "protein": 6,
                  "timesUsed": 2,
                  "unit": "pieces",
                },
              ],
              "total": Object {
                "calories": 533,
                "carbohydrates": 46.5,
                "fat": 18,
                "protein": 41.2,
              },
              "unit": "lb",
              "weight": 201,
            },
          ]
    }
    startAddLog={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "logs": Array [
        Object {
          "date": -154800000,
          "foods": Array [
            Object {
              "amount": 24,
              "calories": 163,
              "carbohydrates": 2.5,
              "createdAt": -1000,
              "fat": 14,
              "id": "3",
              "name": "Almonds",
              "protein": 6,
              "timesUsed": 2,
              "unit": "pieces",
            },
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
            Object {
              "amount": 100,
              "base": Object {
                "amount": 100,
                "calories": 165,
                "carbohydrates": 0,
                "fat": 3.6,
                "protein": 31,
              },
              "calories": 165,
              "carbohydrates": 0,
              "createdAt": 1000,
              "fat": 3.6,
              "id": "2",
              "name": "Chicken Breast",
              "protein": 31,
              "timesUsed": 1,
              "unit": "grams",
            },
          ],
          "total": Object {
            "calories": 533,
            "carbohydrates": 46.5,
            "fat": 18,
            "protein": 41.2,
          },
          "unit": "lb",
          "weight": 202,
        },
        Object {
          "date": 0,
          "foods": Array [
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
          ],
          "total": Object {
            "calories": 410,
            "carbohydrates": 88,
            "fat": 0.8,
            "protein": 8.4,
          },
          "unit": "lb",
          "weight": 200,
        },
        Object {
          "date": 86400000,
          "foods": Array [
            Object {
              "amount": 200,
              "calories": 205,
              "carbohydrates": 44,
              "createdAt": 0,
              "fat": 0.4,
              "id": "1",
              "name": "Rice (Jasmine)",
              "protein": 4.2,
              "timesUsed": 0,
              "unit": "grams",
            },
            Object {
              "amount": 100,
              "base": Object {
                "amount": 100,
                "calories": 165,
                "carbohydrates": 0,
                "fat": 3.6,
                "protein": 31,
              },
              "calories": 165,
              "carbohydrates": 0,
              "createdAt": 1000,
              "fat": 3.6,
              "id": "2",
              "name": "Chicken Breast",
              "protein": 31,
              "timesUsed": 1,
              "unit": "grams",
            },
            Object {
              "amount": 24,
              "calories": 163,
              "carbohydrates": 2.5,
              "createdAt": -1000,
              "fat": 14,
              "id": "3",
              "name": "Almonds",
              "protein": 6,
              "timesUsed": 2,
              "unit": "pieces",
            },
          ],
          "total": Object {
            "calories": 533,
            "carbohydrates": 46.5,
            "fat": 18,
            "protein": 41.2,
          },
          "unit": "lb",
          "weight": 201,
        },
      ],
      "onCreate": [Function],
      "onEdit": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "logs": Array [
          Object {
            "date": -154800000,
            "foods": Array [
              Object {
                "amount": 24,
                "calories": 163,
                "carbohydrates": 2.5,
                "createdAt": -1000,
                "fat": 14,
                "id": "3",
                "name": "Almonds",
                "protein": 6,
                "timesUsed": 2,
                "unit": "pieces",
              },
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
              Object {
                "amount": 100,
                "base": Object {
                  "amount": 100,
                  "calories": 165,
                  "carbohydrates": 0,
                  "fat": 3.6,
                  "protein": 31,
                },
                "calories": 165,
                "carbohydrates": 0,
                "createdAt": 1000,
                "fat": 3.6,
                "id": "2",
                "name": "Chicken Breast",
                "protein": 31,
                "timesUsed": 1,
                "unit": "grams",
              },
            ],
            "total": Object {
              "calories": 533,
              "carbohydrates": 46.5,
              "fat": 18,
              "protein": 41.2,
            },
            "unit": "lb",
            "weight": 202,
          },
          Object {
            "date": 0,
            "foods": Array [
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
            ],
            "total": Object {
              "calories": 410,
              "carbohydrates": 88,
              "fat": 0.8,
              "protein": 8.4,
            },
            "unit": "lb",
            "weight": 200,
          },
          Object {
            "date": 86400000,
            "foods": Array [
              Object {
                "amount": 200,
                "calories": 205,
                "carbohydrates": 44,
                "createdAt": 0,
                "fat": 0.4,
                "id": "1",
                "name": "Rice (Jasmine)",
                "protein": 4.2,
                "timesUsed": 0,
                "unit": "grams",
              },
              Object {
                "amount": 100,
                "base": Object {
                  "amount": 100,
                  "calories": 165,
                  "carbohydrates": 0,
                  "fat": 3.6,
                  "protein": 31,
                },
                "calories": 165,
                "carbohydrates": 0,
                "createdAt": 1000,
                "fat": 3.6,
                "id": "2",
                "name": "Chicken Breast",
                "protein": 31,
                "timesUsed": 1,
                "unit": "grams",
              },
              Object {
                "amount": 24,
                "calories": 163,
                "carbohydrates": 2.5,
                "createdAt": -1000,
                "fat": 14,
                "id": "3",
                "name": "Almonds",
                "protein": 6,
                "timesUsed": 2,
                "unit": "pieces",
              },
            ],
            "total": Object {
              "calories": 533,
              "carbohydrates": 46.5,
              "fat": 18,
              "protein": 41.2,
            },
            "unit": "lb",
            "weight": 201,
          },
        ],
        "onCreate": [Function],
        "onEdit": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
